compiler/tests-common/tests/org/jetbrains/kotlin/codegen/AbstractSteppingTest.kt

```diff
@@ -0,0 +1,211 @@
+
+                        virtualMachine.resume()
+                    }
+                    is ClassPrepareEvent -> {
+                        val methodEntryReq = manager.createMethodEntryRequest()
+                        methodEntryReq.addClassFilter(MAIN_CLASS)
+                        methodEntryReq.setSuspendPolicy(SUSPEND_NONE)
+                        methodEntryReq.enable()
+
+                        val methodExitReq = manager.createMethodExitRequest()
+                        methodExitReq.addClassFilter(MAIN_CLASS)
+                        methodExitReq.setSuspendPolicy(SUSPEND_NONE)
+                        methodExitReq.enable()
+
+                        val stepReq = manager.createStepRequest(event.thread(), StepRequest.STEP_LINE, StepRequest.STEP_INTO)
+                        stepReq.setSuspendPolicy(SUSPEND_NONE)
+                        stepReq.addClassExclusionFilter("java.*")
+                        stepReq.addClassExclusionFilter("sun.*")
+                        stepReq.enable()
+
+                        virtualMachine.resume()
+                    }
+                    is MethodEntryEvent -> {
+                        if (event.location().method().name() == BOX_METHOD) {
+                            shouldRecordSteps = true
+                        }
+                    }
+                    is StepEvent -> {
+                        if (shouldRecordSteps) {
+                            loggedEvents.add(event)
+                        }
+                    }
+                    is MethodExitEvent -> {
+                        if (event.location().method().name() == BOX_METHOD) {
+                            manager.stepRequests().forEach { it.disable() }
+                            shouldRecordSteps = false
+                        }
+                    }
+                    else -> {
+                        throw IllegalStateException("event not handled: $event")
+                    }
+                }
+            }
+        }
+        val actualLineNumbers = loggedEvents
+            .map { event ->
+                "${(event as LocatableEvent).location().method()}:${event.location().lineNumber()}"
+            }
+        TestCase.assertEquals(expectedLineNumbers, actualLineNumbers.joinToString(" "))
+    }
+
+    private fun startDebuggeeProcess(classesDir: File): RunJVMResult {
+        val classpath = listOf(
+            classesDir.absolutePath,
+            PathUtil.getJarPathForClass(Delegates::class.java) // Add Kotlin runtime JAR
+        )
+
+        val command = arrayOf(
+            findJavaExecutable().absolutePath,
+            "-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=127.0.0.1:0",
+            "-ea",
+            "-classpath", classpath.joinToString(File.pathSeparator),
+            MAIN_CLASS
+        )
+
+        val process = ProcessBuilder(*command).start()
+        val port = process.inputStream.bufferedReader().readLine()
```

Comment:
Whould it work properly during parallel test execution? Would it be another port for second process?